File: Problem 3/ans.cpp
#include<iostream>

using namespace std;

int prime(long long int x);

int main()
{
    long long int div=2,max=0,cur,num=600851475143;
    int a;
    cur=num;
    while(cur!=1)
    {
        if (!(cur%div))
        {
            if (prime(div))
                if (div>max)
                    max=div;
            cur/=div;
            div=2;
            continue;
        }
        div++;
    }
    cout << "Largest prime factor of " << num << " is: " << max << endl;
    return 0;
}

int prime(long long int x)
{
    for(long int a=2;a<x;a++)
        if (!(x%a))
            return 0;
    return 1;
}

File: Problem 3/ans.py
def prime(x):
    for i in range(2,x):
        if not x%i:
            return 0
    return 1

def divi(x):
    big=0
    div=2
    while(x!=1):
        if not x%div:
            if prime(x):
                if div>big:
                    big=div
            x//=div
            div=2
            continue
        div+=1
    return big

num=600851475143

print("Largest prime factor of %d is: %d"%(num,divi(num)))